### DISCUSSION TITLE
Goal Animations: Share/Request

### DISCUSSION THREAD

**Discussion by OatmealParmesan** [riffnshred/nhl-led-scoreboard#223] [Show and tell]
With the new goal animation directory (nhl-led-scoreboard/assets/animations/goal), we have new subdirectories for preferred/opposing/general goal animations. 

- Gif file names in these directories do not matter, and if multiple files are present, will be triggered randomly. 
- If you wish to differentiate the preferred/opposing animations, make sure pref_team_only in your config.json is set to false.

Let's share these custom animations! If you have an idea, feel free to submit here as well.

**Comment by OatmealParmesan** [comment]
![goal_light_animation](https://user-images.githubusercontent.com/75916739/109840276-f7e3f000-7c15-11eb-9dc5-e742252523ed.gif)
Rangers - MSG (64x32)

![kick goal_light_animation](https://user-images.githubusercontent.com/75916739/109840379-0fbb7400-7c16-11eb-9ab1-0ab781cef593.gif)
Rangers - Panarin Leg Kick (64x32)

![NYI_goal_light_animation](https://user-images.githubusercontent.com/75916739/109840466-24980780-7c16-11eb-86e4-2d84cbb26e0f.gif)
Fishsticks - Yes! Yes! Yes! (64x32)

![goal_light_animation_gritty](https://user-images.githubusercontent.com/75916739/109906675-f09d0080-7c6e-11eb-895d-dd1d7e302d01.gif)
Flyers - Gritty (64x32)

![fuck](https://user-images.githubusercontent.com/75916739/109906698-f8f53b80-7c6e-11eb-8423-c134c5eb4ad6.gif)
Opposing Goal - #@*! (64x32)

**Reply by riffnshred** [reply]
I'll say it again, those are sick. I was thinking a making an organisation and having a repo for animations and other 3rd party stuff. I'll let you know when its created and you can submit a PR.

**Comment by sflems** [comment]
### For the leafs fans!!!

[carlton_clap.gif](https://user-images.githubusercontent.com/44352296/109903364-8ecd9e00-7c50-11eb-9834-846fb7dbb8a8.gif)
![carlton_clap](https://user-images.githubusercontent.com/44352296/109903364-8ecd9e00-7c50-11eb-9834-846fb7dbb8a8.gif)


[leafs_nation.gif](https://user-images.githubusercontent.com/44352296/109900450-42805f00-7c4c-11eb-9bdf-4b188be801c1.gif)
![leafs_nation](https://user-images.githubusercontent.com/44352296/109900450-42805f00-7c4c-11eb-9bdf-4b188be801c1.gif)

> I'll say it again, those are sick. I was thinking a making an organisation and having a repo for animations and other 3rd party stuff. I'll let you know when its created and you can submit a PR. - @riffnshred 

I could definitely integrate this into the [GUI](https://github.com/sflems/nhl-led-scoreboard-webgui) for saving animations into the various folders. 😉

**Comment by kempdawg20** [comment]
For the Molson fans! (64x32)
![molson4](https://user-images.githubusercontent.com/80005462/109919514-6197e480-7c7e-11eb-9848-227af211297f.gif)

**Comment by ming-lee-5dollar** [comment]
![tuskengoal](https://user-images.githubusercontent.com/80773301/111379738-d9501100-8679-11eb-8929-5d0847f576f6.gif)
Tusken Raider (64x32)
![youcannotdothat](https://user-images.githubusercontent.com/80773301/111379699-cd644f00-8679-11eb-9d1d-dc4815915f0f.gif)
You Can't Do That (64x32)

**Comment by ramsaytc** [comment]
All 64 x 32.
Credit to the people who actually made them...I just resized them.

![avalanche](https://user-images.githubusercontent.com/16809662/112406306-5585cc80-8cea-11eb-9f14-255c48d4064d.gif)
![blackhawks](https://user-images.githubusercontent.com/16809662/112406310-561e6300-8cea-11eb-9419-a094d4f60b31.gif)
![blue jackets](https://user-images.githubusercontent.com/16809662/112406311-561e6300-8cea-11eb-9a3f-3017cc07854e.gif)
![blues](https://user-images.githubusercontent.com/16809662/112406312-561e6300-8cea-11eb-9425-42bc3d1019ed.gif)
![bruins](https://user-images.githubusercontent.com/16809662/112406313-561e6300-8cea-11eb-8f09-6d7ecfe8b8c5.gif)
![canucks](https://user-images.githubusercontent.com/16809662/112406315-56b6f980-8cea-11eb-8931-5102c5b339c4.gif)
![capitals](https://user-images.githubusercontent.com/16809662/112406316-56b6f980-8cea-11eb-821c-c64ae7421dad.gif)
![coyotes](https://user-images.githubusercontent.com/16809662/112406318-56b6f980-8cea-11eb-9d64-7fba20600aef.gif)
![devils](https://user-images.githubusercontent.com/16809662/112406319-574f9000-8cea-11eb-9708-ff8d24b16763.gif)
![ducks](https://user-images.githubusercontent.com/16809662/112406320-574f9000-8cea-11eb-946f-97a6483b1aed.gif)
![flames](https://user-images.githubusercontent.com/16809662/112406321-574f9000-8cea-11eb-82d0-52b1173bdd98.gif)
![flyers](https://user-images.githubusercontent.com/16809662/112406323-574f9000-8cea-11eb-863a-8dbc9547f4d2.gif)
![golden knights](https://user-images.githubusercontent.com/16809662/112406324-574f9000-8cea-11eb-8544-6c3c5ce4c02b.gif)
![habs](https://user-images.githubusercontent.com/16809662/112406325-57e82680-8cea-11eb-9a73-99e521fe26d0.gif)
![hurricanes](https://user-images.githubusercontent.com/16809662/112406326-57e82680-8cea-11eb-945e-d9508f05305d.gif)
![islanders](https://user-images.githubusercontent.com/16809662/112406327-57e82680-8cea-11eb-8082-10eeb0f45b5d.gif)
![jets](https://user-images.githubusercontent.com/16809662/112406328-57e82680-8cea-11eb-95aa-96158a7e642d.gif)
![kings](https://user-images.githubusercontent.com/16809662/112406329-5880bd00-8cea-11eb-9598-5c953f56b8f2.gif)
![kraken](https://user-images.githubusercontent.com/16809662/112406331-5880bd00-8cea-11eb-953e-f65d6a4a88e3.gif)
![lightning](https://user-images.githubusercontent.com/16809662/112406332-5880bd00-8cea-11eb-94cf-fe8ded4d608c.gif)
![maple leafs](https://user-images.githubusercontent.com/16809662/112406333-5880bd00-8cea-11eb-924d-81cb26466345.gif)
![oilers](https://user-images.githubusercontent.com/16809662/112406334-5880bd00-8cea-11eb-816b-b25d1186c4a8.gif)
![panthers](https://user-images.githubusercontent.com/16809662/112406335-59195380-8cea-11eb-881c-7806b0990780.gif)
![penguins](https://user-images.githubusercontent.com/16809662/112406336-59195380-8cea-11eb-864b-53afce030480.gif)
![predators](https://user-images.githubusercontent.com/16809662/112406337-59195380-8cea-11eb-9b00-73e9076e059f.gif)
![rangers](https://user-images.githubusercontent.com/16809662/112406338-59195380-8cea-11eb-8b7f-3ff44b9ea3b0.gif)
![red wings](https://user-images.githubusercontent.com/16809662/112406339-59195380-8cea-11eb-958f-aba3085a41ac.gif)
![sabres](https://user-images.githubusercontent.com/16809662/112406341-59b1ea00-8cea-11eb-888c-99cda03daa85.gif)
![senators](https://user-images.githubusercontent.com/16809662/112406342-59b1ea00-8cea-11eb-8c16-9cee1090a57d.gif)
![sharks](https://user-images.githubusercontent.com/16809662/112406343-59b1ea00-8cea-11eb-85a8-a83cbac8cffd.gif)
![stars](https://user-images.githubusercontent.com/16809662/112406344-59b1ea00-8cea-11eb-9ca6-939e941e1e98.gif)
![wild](https://user-images.githubusercontent.com/16809662/112406345-59b1ea00-8cea-11eb-975b-692b90aa341d.gif)

**Comment by pjockey** [comment]
Do you place the file in /Users/scott/Downloads/nhl-led-scoreboard/assets/animations/goal? and remove the generic one? or how is it called in the script?

**Comment by alexwolf25** [comment]
Does anyone have brief instructions on this? I struggled on getting this to work so I assume others will as well. Much appreciated.

**Reply by pjockey** [reply]
I can tell you to do it, just I'm not near my set up. 

Basically copy the one you want into where the goal animation file is. Rename the goal animation file with an .old extension. Then rename the file of your team to goal_animation_......

**Reply by pjockey** [reply]
I believe when you right click on the one you want and click save image as.. it will provide you a long numbered name, just change that to the name of your team.

Then move that file to the directoy noted below.

pi@raspberrypi:~/nhl-led-scoreboard/assets/animations/goal $ ls
general  goal_light_animation.gif  goal_light_animation.gif.old  opposing  preferred

You'll see I have a goal_light_animation.gif.old. This is the original goal light. Rename your original old or bak and then the one you downloaded and be renamed as follows, example kracken.gif to goal_light_animation.gif. Then when your team scores, you will see the one you downloaded from above.

To copy the file from your computer to the Pi with the scoreboard attached, I did the following from a Mac

scp Flames_score.gif pi@192.168.X.X:nhl-led-scoreboard/assets/animations/goal

**Comment by kempdawg20** [comment]
![shame_penalty2](https://user-images.githubusercontent.com/80005462/141226617-e2ff4364-5966-444f-b409-75a2b47a66db.gif)
Slap Shot Penalty Gif, enjoy... (64x32)

**Comment by nixus619** [comment]
Dumb question time. The goal_light_animation is only inside folder goal and it looks like that goes off only when my team scores. And I see these general, opposing, & preferred folders that are empty. 

What I wanna do is throw all the rangers gifs and randomly choose between the ones I downloaded from the threads above & the standard goal_light_animation gif. So I just put them direction in the goal folder. Will this give me a random choice of the 4 I have in there? Thanks

**Reply by falkyre** [reply]
So the way the code works is if you put animations into either of those folders, then it will randomly select which animation to play. 
 If you select in the config.json that you want to show only your preferred team goal animation then it will only look for animations in the preferred folder and randomly play those.  If you want to just randomly play an animation for any goal, put all of your animations into the general folder. 

So you can have this as a possibility:

1.  Put all animations into the general folder.  The code will randomly play the animations in that folder for either team
2. Put animations into preferred and into opposing.  The code will randomly play an animation for your preferred team's goals and the opposing team's goals.  So you can have different animations for the team you like and the teams you don't
3. Only put animations into the preferred folder, select to play animations for preferred team only in the config.json and you will only get animations when the team you like scores

The line in the config.json that controls this looks like:
`"goal_animations": {
            "pref_team_only": true
        }`

**Reply by nixus619** [reply]
Awesome, this makes more sense.

Thanks so much!

**Reply by nixus619** [reply]
Back with another question. I use a smart plug to turn on and off the board (I don't want it to run all day and waste electricity when I'm not home). I notice that if I add the gifs and then check the next day, it's back to the default... I'm wondering if this is because I'm not doing a normal shut down of the board (basically just pulling the plug). You think that could have an impact?

**Reply by pjockey** [reply]
I would think it is because they are not getting into the correct directory. I the goal gif for my team and it works on every reboot. As well, I have a legacy or different logo and it too does not change. They did a change to the config.json.schema a while back that is suppose to not revert back to the originals if the gifs/jpgs are place in the correct direct and the config file is use properly.

I'm trying to think what all was involved.  A discussion should be here somewhere.

Have searched the Discord channel?
